src/Dispatcher.php

```diff
@@ -47,8 +47,11 @@ public function dispatch(ServerRequestInterface $request, RequestHandlerInterfac
 
     public function withMiddlewares(array $middlewares): DispatcherInterface
     {
+        $stack = $this->stack;
+        $this->stack = null;
         $clone = clone $this;
         $clone->middlewares = $middlewares;
+        $this->stack = $stack;
```

Comment:
Isn't it equivalent?
```suggestion
        $clone = clone $this;
        $clone->middlewares = $middlewares;
        $clone->stack = null;
```

Reply:
> Isn't it equivalent?

I this case you should change modificator of `stack` property to `protect`.

Reply:
Though.... I'm wrong, the private modifier works too.

Reply:
Done. Thanks.